MergerLogic/ImageProcessing/TileMerger.cs

```diff
@@ -158,7 +164,9 @@ private void AddTileToImageList(Coord targetCoords, Tile? tile, List<MagickImage
             tileImage = new MagickImage(tileBytes);
             if (tile.Z < targetCoords.Z)
             {
-                this._tileScaler.Upscale(tileImage, tile, targetCoords);
+                var upscale = this._tileScaler.Upscale(tileImage, tile, targetCoords);
```

Comment:
why don't you dispose and replace the "tileImage" inside the method? every one that uses this method needs to remember to dispose it himself - can lead to bugs

Reply:
1. replacing inside the function requires making the pram ref.
2. what if the calling function still want to use the original tile as well as the upscaled? disposing should be handler in the same level as the allocation